live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/hx_insight/storage/connector/HttpFileUploader.java

```diff
@@ -98,6 +98,17 @@ public void configure()
     @Override
     @SuppressWarnings({"PMD.CloseResource", "PMD.PreserveStackTrace"})
     public void upload(FileUploadRequest fileUploadRequest)
+    {
+        upload(fileUploadRequest, null);
+    }
+
+    @Retryable(retryFor = EndpointServerErrorException.class,
+            maxAttemptsExpression = "#{@integrationProperties.hylandExperience.storage.upload.retry.attempts}",
+            backoff = @Backoff(delayExpression = "#{@integrationProperties.hylandExperience.storage.upload.retry.initialDelay}",
+                    multiplierExpression = "#{@integrationProperties.hylandExperience.storage.upload.retry.delayMultiplier}"))
+    @Override
+    @SuppressWarnings({"PMD.CloseResource", "PMD.PreserveStackTrace"})
+    public void upload(FileUploadRequest fileUploadRequest, String nodeId)
```

Comment:
I think we should _only_ have this method, and get rid of the single parameter method. If this means updating a load of unit tests then that's fine - the tests should ideally call the same methods that the production code calls anyway.

Reply:
done in [260c2e1](https://github.com/Alfresco/hxinsight-connector/pull/274/commits/260c2e120a8371465126244b8ca2362e94ece209)